File: android/app/src/main/kotlin/com/wagmilabs/sigil/ui/implementorinit/ApprovalRequestView.kt
package com.wagmilabs.sigil.ui.implementorinit

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material.icons.filled.Key
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.wagmilabs.sigil.network.models.ImplementorInitRequest
import java.text.SimpleDateFormat
import java.util.*

@Composable
fun ApprovalRequestView(
    request: ImplementorInitRequest,
    timeRemaining: Int,
    onApprove: () -> Unit,
    onReject: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(24.dp)
    ) {
        Icon(
            imageVector = Icons.Default.Key,
            contentDescription = null,
            modifier = Modifier.size(64.dp),
            tint = MaterialTheme.colorScheme.primary
        )

        Text(
            text = "Mnemonic Request",
            style = MaterialTheme.typography.titleLarge
        )

        Card {
            Column(modifier = Modifier.padding(16.dp)) {
                DetailRow("Implementor:", request.implementorName)
                Spacer(modifier = Modifier.height(8.dp))
                DetailRow("Identifier:", request.implementorId)
                Spacer(modifier = Modifier.height(8.dp))
                DetailRow("Requested:", formatDate(request.timestamp))
            }
        }

        Text(
            text = "A new 24-word mnemonic will be generated and sent to the implementor. You will need to write it down before submission.",
            style = MaterialTheme.typography.bodySmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center
        )

        if (timeRemaining < 60) {
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                Icon(Icons.Default.AccessTime, contentDescription = null)
                Text("Expires in ${timeRemaining}s")
            }
        }

        Row(horizontalArrangement = Arrangement.spacedBy(16.dp)) {
            OutlinedButton(onClick = onReject) { Text("Reject") }
            Button(onClick = onApprove) { Text("Approve") }
        }
    }
}

@Composable
private fun DetailRow(label: String, value: String) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(label, fontWeight = FontWeight.Medium)
        Text(value, color = MaterialTheme.colorScheme.onSurfaceVariant)
    }
}

private fun formatDate(date: Date): String {
    val formatter = SimpleDateFormat("MM/dd/yy hh:mm a", Locale.getDefault())
    return formatter.format(date)
}

File: ios/Sources/UI/ImplementorInit/ApprovalRequestView.swift
import SwiftUI

public struct ApprovalRequestView: View {
    let request: ImplementorInitRequest
    let timeRemaining: Int
    let onApprove: () -> Void
    let onReject: () -> Void

    public init(
        request: ImplementorInitRequest,
        timeRemaining: Int,
        onApprove: @escaping () -> Void,
        onReject: @escaping () -> Void
    ) {
        self.request = request
        self.timeRemaining = timeRemaining
        self.onApprove = onApprove
        self.onReject = onReject
    }

    public var body: some View {
        VStack(spacing: 24) {
            Image(systemName: "key.fill")
                .font(.system(size: 64))
                .foregroundColor(.blue)

            Text("Mnemonic Request")
                .font(.title2)
                .fontWeight(.semibold)

            GroupBox {
                VStack(alignment: .leading, spacing: 12) {
                    DetailRow(label: "Implementor:", value: request.implementorName)
                    DetailRow(label: "Identifier:", value: request.implementorId)
                    DetailRow(label: "Requested:", value: formatDate(request.timestamp))
                }
            }

            Text("A new 24-word mnemonic will be generated and sent to the implementor. You will need to write it down before submission.")
                .font(.footnote)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)

            if timeRemaining < 60 {
                HStack {
                    Image(systemName: "clock")
                    Text("Expires in \(timeRemaining)s")
                }
                .font(.caption)
                .foregroundColor(.orange)
            }

            HStack(spacing: 16) {
                Button("Reject") {
                    onReject()
                }
                .buttonStyle(.bordered)

                Button("Approve") {
                    onApprove()
                }
                .buttonStyle(.borderedProminent)
            }
        }
        .padding()
    }

    private func formatDate(_ date: Date) -> String {
        let formatter = DateFormatter()
        formatter.dateStyle = .short
        formatter.timeStyle = .short
        return formatter.string(from: date)
    }
}

struct DetailRow: View {
    let label: String
    let value: String

    var body: some View {
        HStack {
            Text(label)
                .fontWeight(.medium)
            Spacer()
            Text(value)
                .foregroundColor(.secondary)
        }
    }
}
